### PR TITLE
Remove unnecessary injected error logging in crash test

### PR DESCRIPTION
**Pull request by hx235**
Context/Summary: as titled, since injected error log isn't that useful for debugging and takes up a lot of console printing space

Test: CI

### CODE REVIEW

File: db_stress_tool/db_stress_test_base.cc [hunk 1]
```diff
@@ -1704,10 +1698,13 @@ Status StressTest::TestIterateImpl(ThreadState* thread,
       op_logs += "S " + key.ToString(true) + " ";
     }
 
-    if (IsRetryableInjectedError(iter->status()) ||
-        IsRetryableInjectedError(cmp_iter->status())) {
-      return !iter->status().ok() ? iter->status() : cmp_iter->status();
+    if (!iter->status().ok() && IsRetryableInjectedError(iter->status())) {
```
Comment (pdillinger) [inline db_stress_tool/db_stress_test_base.cc:1701]: From the name, I would assume `IsRetryableInjectedError(s)` implies `!s.ok()`. Should we make that assumption / simplification?

File: db_stress_tool/db_stress_test_base.h [hunk 2]
```diff
@@ -281,6 +281,7 @@ class StressTest {
   }
 
   bool IsRetryableInjectedError(const Status& s) const {
+    assert(!s.ok());
```
Comment (pdillinger) [inline db_stress_tool/db_stress_test_base.h:284]: I guess it's not compatible with this assertion. Was this assertion useful in getting the logic correct?

**Review by pdillinger — APPROVED** [review]
Overall LGTM. Thanks for the cleanup!

**Review by jowlyzhang — APPROVED** [review]
LGTM, thanks for the improvement!

File: db_stress_tool/db_stress_test_base.h (hunk 2 above)
Comment (pdillinger) [reply db_stress_tool/db_stress_test_base.h:284]: A more technically correct name with this assertion would be `IsErrorInjectedAndRetryable`

Comment (hx235) [reply db_stress_tool/db_stress_test_base.h:284]: Will fix this thanks!

File: db_stress_tool/db_stress_test_base.cc (hunk 1 above)
Comment (hx235) [reply db_stress_tool/db_stress_test_base.cc:1701]: yes

Comment (hx235) [reply db_stress_tool/db_stress_test_base.cc:1701]: We fix the name so it's obvious that the parameter s is an error

File: db_stress_tool/db_stress_test_base.h (hunk 2 above)
Comment (hx235) [reply db_stress_tool/db_stress_test_base.h]: Fixed

File: db_stress_tool/db_stress_test_base.cc (hunk 1 above)
Comment (hx235) [reply db_stress_tool/db_stress_test_base.cc]: Fixed

**Pull request closed without merging**